src/training/train.py

```
@@ -1,443 +1,467 @@
+            restore_best_weights=True
+        )
+        
+        # Training history
+        self.history: Dict[str, List[float]] = {
+            'train_loss': [],
+            'val_loss': [],
+            'epoch_times': [],
+            'learning_rates': []
+        }
+        
+        logger.info(f"Trainer initialized:")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)